core-bundle/src/Resources/contao/library/Contao/Files.php

```diff
@@ -156,6 +159,8 @@ public function fopen($strFile, $strMode)
 	{
 		$this->validate($strFile);
 
+		$this->mkdir($this->strRootDir);
```

Comment:
```suggestion
		$this->mkdir(\dirname($this->strRootDir . '/' . $strFile));
```
Did you mean something like this?

Reply:
But `$this->mkdir()` is not recursive so maybe better to use something like this?
```suggestion
		(new Filesystem())->mkdir(\dirname($this->strRootDir . '/' . $strFile));
```

Reply:
Yes, definitely use the `Filesystem` component for mkdir.